hydra/grammar/OverrideParser.g4

```diff
@@ -47,12 +47,17 @@ function: ID POPEN (argName? element (COMMA argName? element )* )? PCLOSE;
 
 // Data structures.
 
-listValue: BRACKET_OPEN                          // [], [1,2,3], [a,b,[1,2]]
-    (element(COMMA element)*)?
-BRACKET_CLOSE;
-
+listValue: BRACKET_OPEN sequence? BRACKET_CLOSE;                          // [], [1,2,3], [a,b,[1,2]]
 dictValue: BRACE_OPEN (dictKeyValuePair (COMMA dictKeyValuePair)*)? BRACE_CLOSE;  // {}, {a:10,b:20}
-dictKeyValuePair: ID COLON element;
+dictKeyValuePair: (ID | interpolation) COLON element;
```

Comment:
I don't think interpolation in a key makes sense. can you show a use case?

Reply:
Well I don't see a reason why it would make more sense to allow interpolations in values compared to keys, for a start.

First thing I can think of (probably not the most complete/simple/convincing), let's say there are different folders, each containing a known type of files, and each file must be processed by a function that depends on its type. And let's say we want to perform some kind of reduce operation over the processed files, whose output is saved in the config variable `reduced`. It may look like this:

```yaml
images:
    path: /path/to/images
    process_function: process_image
videos:
    path: /path/to/videos
    process_function: process_video
captions:
    path: /path/to/captions
    process_function: process_caption
reduced: ${process:{${images.path}: ${images.process_function}, ${videos.path}: ${videos.process_function}, ${captions.path}: ${captions.process_function}}}
```

The code for the `process` resolver would be something like:

```python
def process(**to_process):
    return some_reduce_operation(f(os.path.join(path, x)) for path, f in to_process.items() for x in os.listdir(path))
```

Reply:
Okay, I will comment on the other thread. (easier to find).